pkg/controller/registry/reconciler/grpc.go

```diff
@@ -274,6 +283,33 @@ func (c *GrpcRegistryReconciler) ensureService(source grpcCatalogSourceDecorator
 	return err
 }
 
+// ServiceHashMatch will check the hash info in existing Service to ensure its
+// hash info matches the desired Service's hash.
+func ServiceHashMatch(existing, new *corev1.Service) bool {
+	labels := existing.GetLabels()
+	if len(labels) == 0 {
+		return false
+	}
+	existingSvcSpecHash, ok := labels[ServiceHashLabelKey]
+	if !ok {
+		return false
+	}
+
+	hash, _ := new.GetLabels()[ServiceHashLabelKey]
```

Comment:
should we check that the key exists here (as above) or are we saying we will always have a hash on the `new` service? That would be an error based on this logic.

Reply:
The `new` Service is coming from `source.Service()` which with the code change will guarantee there is a label. However, I can see your point that this func may be used else where so the provided Service may not have labels. I can change that.